### PR TITLE
Issue 13030: Wait for HTTP to allow ACME startup

### LINKED ISSUE #9017: SEC 27 - Support for ACME certificate management and LetsEncrypt
LetsEncrypt has become the most popular free CA signing site. There are now over 100 million certs signed by LetsEncrypt in the wild today. This makes it one of the largest CAs. New cloud native applications/microservices would be served well by adding support for the underlying protocol that LetsEncrypt uses - called ACME (Automated Certificate Management Environment) which is fast becoming the defacto standard. With this feature we plan to add ACME support to Open Liberty.

**Link to WAD/UFO**
https://ibm.box.com/s/y2cjatj56o00rlrx34ed4omr6xgxb0hu

https://ibm.ent.box.com/file/667631862611

**Direct Link to Feature Checklist (gets hidden in the comments)**
https://github.com/OpenLiberty/open-liberty/issues/9017#issuecomment-571420951

 **Issue Queries:**
[High priority open issues](https://github.com/OpenLiberty/open-liberty/issues?q=is%3Aopen+type%3Aissue+label%3Apriority%2Fhigh+acmeCA-2.0+in%3Atitle+) (Min viable product for a beta)
[Medium priority open issues](https://github.com/OpenLiberty/open-liberty/issues?q=is%3Aopen+type%3Aissue+label%3Apriority%2Fmedium+acmeCA-2.0+in%3Atitle+)
[Unlabeled open issuse](https://github.com/OpenLiberty/open-liberty/issues?q=is%3Aopen+type%3Aissue+acmeCA-2.0+in%3Atitle++-label%3Apriority%2Fmedium++-label%3Apriority%2Fhigh)
[All open issues](https://github.com/OpenLiberty/open-liberty/issues?q=is%3Aopen+type%3Aissue+acmeCA-2.0+in%3Atitle)

**Shortcut links**
Doc/Blog prep issue is #9869
Doc request is here https://github.com/OpenLiberty/docs/issues/1699
Feature Test Summary is #9850
SVT test prep is [here](https://github.ibm.com/was-liberty/WS-CD-Open/issues/20744)
STE in progress is https://github.ibm.com/kristip/site/blob/patch-1/_STE/WAS/acmeCA-2.0
Published STE is https://pages.github.ibm.com/WASL3/site/STE/WAS/acmeCA-2.0/
GA Blog issue https://github.com/OpenLiberty/open-liberty/issues/12860

For internal reference: Demo for EOI on June 2 (20.11). Demo starts at 23 min mark. https://ibm.ent.box.com/file/673740303546

### LINKED ISSUE #13030: acmeCA-2.0: Wait for HTTP port to open before starting ACME flow
Previously, we were waiting for the internal ACME servlet to open. When an application takes longer to start, the HTTP port may not be open. We need to delay until the HTTP port opens so we can receive the callback from the ACME CA.

For #9017

### PR DESCRIPTION
**Pull request by kristip17**
Fixes #13030 

- Wait on both the ACME web app and HTTP opening
- Add internal config for timeout
- Normalize minRenew config
- Added test with "slow" app to make sure HTTP is delayed and ACME waits appropriately
- Remove app1 from acme_fat, not needed

### CODE REVIEW

**Comment by kristip17** [comment]
#build

**Comment by kristip17** [comment]
#libby-bot

File: dev/com.ibm.ws.security.acme/src/com/ibm/ws/security/acme/internal/AcmeApplicationStateListener.java [hunk 1]
```diff
@@ -140,10 +196,51 @@ public void waitUntilWebAppAvailable() throws AcmeCaException {
...
+
+				/*
+				 * If the wait above expired and we weren't signaled by the startedPhase2
+				 * method, we will attempt to start ACME anyway. If the HTTP port is truly not
+				 * open, then the CA call back to the well-known URL will fail
+				 * ("Connection refused" or similar).
+				 */
+				if (!signaled || !isHttpStarted) {
+					if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+						Tr.event(tc, methodName
```
[the remark this answers is not in the document]
Comment (kristip17) [reply dev/com.ibm.ws.security.acme/src/com/ibm/ws/security/acme/internal/AcmeApplicationStateListener.java:242]: Added a warning message.

**Comment by andrea-pichler** [comment]
Completed the requested L2 message review

**Comment by kristip17** [comment]
#run-libby-bot

**Comment by kristip17** [comment]
This is ready for message review.

**Comment by kristip17** [comment]
#run-libby-bot

File: dev/com.ibm.ws.security.acme/resources/com/ibm/ws/security/acme/resources/AcmeMessages.nlsprops [hunk 2]
```diff
@@ -179,9 +179,9 @@ CWPKI2035E=CWPKI2035E: The ACME service could not store the signed certificate i
...
-CWPKI2036E.explanation=The application used to complete a certificate request did not start in the expected time frame. A signed certificate cannot be requested.
-CWPKI2036E.useraction=Review the log for earlier messages or errors.
+CWPKI2036W=CWPKI2036W: The ACME service timed out waiting for the web authorization application to start. The application is required to complete a certificate request with an ACME certificate authority. The certificate request is attempted. The service waited for {0}.
+CWPKI2036W.explanation=The application used to complete a certificate request did not start in the expected time frame. If the application starts, the request proceeds. If the application does not start, the certificate request fails.
+CWPKI2036W.useraction=Review the log for earlier messages or errors. Review the log for a CWWKT0016I message that includes a web application with the ''acme-challenge'' URL which indicates the internal application started.
```
Comment (helyarp) [inline dev/com.ibm.ws.security.acme/resources/com/ibm/ws/security/acme/resources/AcmeMessages.nlsprops:184]: A nit:  Change "URL which indicates the internal application started." to "URL to indicate that the internal application started." or to "URL, which indicates that the internal application started."

Comment (kristip17) [reply dev/com.ibm.ws.security.acme/resources/com/ibm/ws/security/acme/resources/AcmeMessages.nlsprops]: Changed

**Comment by kristip17** [comment]
#run-libby-bot

**Pull request merged**